temporalio/worker/_activity.py

```diff
@@ -436,6 +438,7 @@ async def _run_activity(
                     runtime_metric_meter=None
                     if sync_non_threaded
                     else self._metric_meter,
+                    cancellation_details=running_activity.cancellation_details,
```

Comment:
The context says it is optional, but this is set with a never-`None` value. I think you may need some kind of "holder" wrapper.

Reply:
That would be preferred over a non-optional field?
The indirection with a holder seems a little clunky